WalletWasabi/Interfaces/IDeepEqual.cs

```diff
@@ -0,0 +1,7 @@
+namespace WalletWasabi.Interfaces;
+
+public interface IDeepEqual<T>
```

Comment:
We have always tried, and I think we should still do, to avoid abstractions that are not absolutely necessary. I think this is not the case.